src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -308,12 +312,11 @@ public override StackValue Execute(ProtoCore.Runtime.Context c, List<StackValue>
                         // create language blocks for true and false branch, 
                         // so directly return the value.
                         if (runtimeCore.Options.GenerateSSA)
-                            return svCondition.RawBooleanValue ? svTrue : svFalse;
+                            return svCondition.BooleanValue ? svTrue : svFalse;
 
                         Validity.Assert(svTrue.IsInteger);
                         Validity.Assert(svFalse.IsInteger);
-                        int blockId = (1 == (int)svCondition.opdata) ? (int)svTrue.opdata : (int)svFalse.opdata;
-
+                        int blockId = svCondition.BooleanValue ? (int)svTrue.IntegerValue : (int)svFalse.IntegerValue;
```

Comment:
This is one such hard-casting area that we can avoid if `IntegerValue` is made `int` instead of `long`. Same for the one below.